File: activityhookdemo/src/main/java/com/zero/activityhookdemo/MainActivity.java
package com.zero.activityhookdemo;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.zero.activityhookdemo.hook.HookHelper;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "Zero";

    @BindView(R.id.tv_show)
    TextView tvShow;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);
    }

    @OnClick(R.id.btn_hook1)
    public void onBtnHook1Clicked() {

    }

    @OnClick(R.id.btn_hook2)
    public void onBtnHook2Clicked() {

        File file =new File("a.txt");

        try {
            FileInputStream inputStream = new FileInputStream(file);
//            inputStream.read();
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

    }

    @OnClick(R.id.btn_hook3)
    public void onBtnHook3Clicked() {

        Intent intent = new Intent(this,TargetActivity.class);
        startActivity(intent);
    }

    @OnClick(R.id.btn_hook4)
    public void onBtnHook4Clicked() {
        Log.i(TAG, "onBtnHook4Clicked: " + Build.VERSION.SDK_INT);
        HookHelper.hookAMSAidl();
        HookHelper.hookHandler();
        Intent intent = new Intent(this,TargetActivity.class);
        startActivity(intent);
    }
}

File: viewpager/src/main/java/com/donfyy/viewpager/MainActivity.kt
package com.donfyy.viewpager

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.donfyy.viewpager.lazyloading.LazyMainActivity
import com.donfyy.viewpager.lazyloading.lazy2.LazyMain2Activity
import com.donfyy.viewpager.lazyloading.lazy3.LazyMain3Activity
import com.donfyy.viewpager.wrap.WrapActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onEnterMain1(view: View) {
        startActivity(Intent(this, LazyMainActivity::class.java))
    }
    fun onEnterMain2(view: View) {
        startActivity(Intent(this, LazyMain2Activity::class.java))
    }
    fun onEnterMain3(view: View) {
        startActivity(Intent(this, LazyMain3Activity::class.java))
    }
    fun onEnterWrapMain(view: View) {
        startActivity(Intent(this, WrapActivity::class.java))
    }
}